### PR TITLE
chore: [VUMM-760] Remove organization for permission v2

### PR DESCRIPTION
**Pull request by evgeniilezhnin**
## Impact and Context

The concept of "organizations"—as they existed pre-permission v2—is being replaced with "workspaces". This PR refactors tests to use v2 workspaces instead of organizations.

## Risks and Area of Effect

This should have no impact for users. New clients can continue to be used with pre-permission v2 platforms.

`Connection.get_workspace_name_from_legacy_id()` is not needed because every entity has been using the newer workspace service for at least two years (https://github.com/VertaAI/modeldb/pull/1914/files#r566375578).

Organization CRUD in the client was restricted to tests, so it won't impact publicly-supported user code.

## Testing
- [x] Unit test
- [ ] Deployed to dev env
- [ ] Other (explain)

Test suite passes against a v1 environment ([Slack thread](https://vertaai.slack.com/archives/C03PWNZU5CM/p1674775415283039?thread_ts=1674666909.564999&cid=C03PWNZU5CM)).

Evgenii also stated

> verified it by running test_permission, test_endpoint and registry test packages

## Reverting
- [ ] Contains Migration - _Do Not Revert_

### CODE REVIEW

**Review by liuverta — COMMENTED** [review]
Let me know if you encounter issues with specific tests. Some of them may be worth removing because they are very old.

File: client/verta/tests/conftest.py
Comment (liuverta) [inline client/verta/tests/conftest.py:121]: 👍 I am very glad to get rid of this 🔥

File: client/verta/verta/client.py
Comment (liuverta) [inline client/verta/verta/client.py:1745]: Unfortunately, I realize that a lot of our tests right now depend on creating custom workspace during tests 🤔 

I will need to talk to Elizabeth about whether we can remove these specific Python tests in the future, but for now I think maybe we need a `_create_workspace()`, `Workspace.add_member()` and `Workspace.delete()` just for the tests

**Review by liuverta — COMMENTED** [review]
Just to let you know: When these tests run in Jenkins, `client` is a sys admin account, so maybe we can assume it has permissions to create and delete workspaces.

File: client/verta/verta/client.py
Comment (evgeniilezhnin) [reply client/verta/verta/client.py:1745]: Is it necessary to create a workspace during test? Why is it a problem to use existing workspace?

It is possible to create a workspace during test, but the problem is that we don't have workspace.add_member. Members are added to workspaces using groups. So the problem is that if we create workspaces in test we also need to call endpoints to access or create groups, roles, users and organizations

Comment (evgeniilezhnin) [reply client/verta/verta/client.py:1745]: Also please note that in permission v2 we can't set individual permission by resource id. The system is simplified comparing to v1 so now admin can set permission to use resource types by groups of users in specified workspaces.
So as I see some of permission tests in Python Client can't be migrated to v2.

**Comment by liuverta** [comment]
The client tests should be able to run against any new, fresh dev
environment. If the workspace should not be created when the tests run,
then I am not sure when it will be created at all.

I would prefer implementing the minimum necessary client functions to
create a workspace and group for the tests to use. If that isn't possible,
I will ask Elizabeth if these tests can just be removed.

**Comment by evgeniilezhnin** [comment]
> The client tests should be able to run against any new, fresh dev environment. If the workspace should not be created when the tests run, then I am not sure when it will be created at all. I would prefer implementing the minimum necessary client functions to create a workspace and group for the tests to use. If that isn't possible, I will ask Elizabeth if these tests can just be removed.

actually there is always a default workspace Default that is created during a migration. We can use it, but anyway it is not a problem to create a new besides that. So it is possible.
There is a second separate problem that there is no more visibility for individual resources. That one is not possible I think.

**Comment by liuverta** [comment]
Hmmmm I see. Maybe I am misremembering the tests, and the default workspace
is okay to use for most of them 🤔 In that case I will leave it to your
judgment, sorry for any confusion!

Yes, I remember there were some per-entity `visibility` and maybe even
`public_within_org` tests. I agree those may simply need to be removed for
v2.

**Comment by evgeniilezhnin** [comment]
I'm planning to add workspace creation because I think it will be useful at least in a couple of tests where we can use private workspace instead of private resource type, also it can be useful for cleanup

**Comment by evgeniilezhnin** [comment]
I'm currently trying to test workspaces creation: https://github.com/VertaAI/modeldb/pull/3439
But at the moment there is a problem with permission v2 proxy: https://vertaai.slack.com/archives/C03PWNZU5CM/p1670581917882709?thread_ts=1669730220.761939&cid=C03PWNZU5CM

In the current PR tests use default workspace where possible. I verified it by running test_permission and registry test packages for now

File: client/verta/verta/_internal_utils/_utils.py
Comment (liuverta) [inline client/verta/verta/_internal_utils/_utils.py]: Could you please help rename this function to `_get_visible_workspaces()`? 🙇‍♂️

File: client/verta/verta/_internal_utils/_utils.py
Comment (liuverta) [inline client/verta/verta/_internal_utils/_utils.py]:
```suggestion
    def _get_organization_id(self):
```

File: client/verta/tests/test_permissions/test_workspace.py
Comment (liuverta) [inline client/verta/tests/test_permissions/test_workspace.py:69]: fyi, this test is very important. We should add it again when workspace creation is implemented.

File: client/verta/tests/registry/model_version/test_crud.py
Comment (liuverta) [inline client/verta/tests/registry/model_version/test_crud.py:193]: If possible, I do not think we should be removing any of the `pytest.raises(requests.HTTPError, match="^403")` checks in this file. They seem important to make sure non-admins cannot change lock level. Is it possible to keep them, or bring them back after we implement workspace creation?

(by the way: when tests run in Jenkins, `client` is admin and `client_2` is non-admin)

File: client/verta/tests/test_permissions/test_sharing_old.py
Comment (liuverta) [inline client/verta/tests/test_permissions/test_sharing_old.py]: Does the [`ORG_SCOPED_PUBLIC` concept](https://github.com/VertaAI/protos-all/blob/4f8323557c43ad324456bb8ee044a6169a87933b/protos/protos/public/common/CommonService.proto#L162-L173) still exist in our platform? 🤔 if not, we can probably just remove this whole file.

File: client/verta/tests/test_permissions/test_visibility_e2e.py
Comment (liuverta) [inline client/verta/tests/test_permissions/test_visibility_e2e.py:19]: @evgeniilezhnin Instead of deleting this file completely, could we add

```python
pytest.skip("visibility deprecated; tests disabled for permissions v2", allow_module_level=True)
```

? I'd like to check later to see if we can adjust these tests for v2.

File: client/verta/tests/test_endpoint/test_endpoint.py
Comment (liuverta) [inline client/verta/tests/test_endpoint/test_endpoint.py]:
```suggestion
        workspace,
```

File: client/verta/tests/test_dataset_versioning/test_dataset_version.py
Comment (liuverta) [inline client/verta/tests/test_dataset_versioning/test_dataset_version.py]: Should this be removed? I'm not sure if it is a part of the test, or if it was for debugging.

File: client/verta/verta/_internal_utils/_utils.py
Comment (liuverta) [inline client/verta/verta/_internal_utils/_utils.py:302]: Just to confirm @evgeniilezhnin: Are you sure we no longer need this? For context, this was because projects and datasets used an older ID system from before WorkspaceService existed.

Are we migrating project and dataset workspace IDs to the new permissions v2 workspace IDs?

**Review by liuverta — CHANGES_REQUESTED** [review]
I have some feedback and questions.

Could you also add more details about how you tested this in the PR description under **Testing**?

Also, do you know if we should wait until permissions v2 is deployed to dev.verta.ai before merging this?

File: client/verta/verta/_internal_utils/_utils.py
Comment (evgeniilezhnin) [reply client/verta/verta/_internal_utils/_utils.py:302]: when changing from permissions v1 to permissions v2 there is no workspace IDs format change. We changed ids format a long time ago. After that we refactored services to store resource-workspace information in UAC only in the new format. So now it should not be stored in old format anywhere and migration from old workspace ids format to new format should be already executed everywhere

File: client/verta/verta/_internal_utils/_utils.py
Comment (evgeniilezhnin) [reply client/verta/verta/_internal_utils/_utils.py]: renamed. This change was in personal workspace removal PR, but can add it here too

File: client/verta/verta/_internal_utils/_utils.py
Comment (evgeniilezhnin) [reply client/verta/verta/_internal_utils/_utils.py]: renamed

File: client/verta/tests/test_permissions/test_workspace.py
Comment (evgeniilezhnin) [reply client/verta/tests/test_permissions/test_workspace.py:69]: reverted removal. I think it can be added here

File: client/verta/tests/registry/model_version/test_crud.py
Comment (evgeniilezhnin) [reply client/verta/tests/registry/model_version/test_crud.py:193]: I think workspace creation is needed for this test, I couldn't make it work with a default workspace. So I can add these checks again after workspace creation implemented in client

File: client/verta/tests/test_permissions/test_sharing_old.py
Comment (evgeniilezhnin) [reply client/verta/tests/test_permissions/test_sharing_old.py]: no, in v2 we can't set individual resources permissions. Removed the test

File: client/verta/tests/test_dataset_versioning/test_dataset_version.py
Comment (evgeniilezhnin) [reply client/verta/tests/test_dataset_versioning/test_dataset_version.py]: yes, removed it now

File: client/verta/tests/test_permissions/test_visibility_e2e.py
Comment (evgeniilezhnin) [reply client/verta/tests/test_permissions/test_visibility_e2e.py:19]: added

**Comment by evgeniilezhnin** [comment]
> I have some feedback and questions.
> 
> Could you also add more details about how you tested this in the PR description under **Testing**?
> 
> Also, do you know if we should wait until permissions v2 is deployed to dev.verta.ai before merging this?

Added

yes, I think permission v2 flag should be enabled before merging client change on dev.verta.ai, testsqlserver, testregular

**Comment by evgeniilezhnin** [comment]
tried running the test from this branch for testregular (after deploy), but there are authorization problems which were also there in previous runs: https://jenkins.dev.verta.ai/job/client/job/python/job/pytest-3.7/388/. Now I'm trying to run test for testsqlserver

**Comment by evgeniilezhnin** [comment]
when testing on testsqlserver there is a problem that default workspace is not set. I think it can be fixed either by implementing migration on UAC or by using custom workspaces in Client which is the thing I'm working on now

**Comment by evgeniilezhnin** [comment]
> when testing on testsqlserver there is a problem that default workspace is not set. I think it can be fixed either by implementing migration on UAC or by using custom workspaces in Client which is the thing I'm working on now

migration helped, but there are still test issues for testsqlserver, that I will fix

File: client/verta/tests/registry/model_version/test_crud.py
Comment (evgeniilezhnin) [reply client/verta/tests/registry/model_version/test_crud.py:193]: reverted here: https://github.com/VertaAI/modeldb/pull/3439/files

**Comment by evgeniilezhnin** [comment]
> when testing on testsqlserver there is a problem that default workspace is not set. I think it can be fixed either by implementing migration on UAC or by using custom workspaces in Client which is the thing I'm working on now

I think it is better to merge the current change when https://github.com/VertaAI/modeldb/pull/3439/files is approved

File: client/verta/verta/_internal_utils/_utils.py
Comment (liuverta) [inline client/verta/verta/_internal_utils/_utils.py:281]: Is it possible for `getVisibileWorkspaces` to return workspaces in other organizations? `list(workspace_names)[0]` seems like it will basically return the `org_id` of a random workspace, so I'm worried that if we have a multi-organization setup, we might get the wrong ID 🤔 but I'm also not sure how permissions v2 works 😅

File: client/verta/verta/_internal_utils/_utils.py
Comment (liuverta) [inline client/verta/verta/_internal_utils/_utils.py]: fyi @evgeniilezhnin, I had been talking with Elizabeth about changing UAC so that the client (and client users) would not need to worry about `org_id`, and instead only use workspace name.

File: client/verta/verta/_internal_utils/_utils.py
Comment (liuverta) [inline client/verta/verta/_internal_utils/_utils.py]: See my other comment about maybe not needing `org_id`: https://github.com/VertaAI/modeldb/pull/3407/files#r1067587726

File: client/verta/verta/tracking/entities/_experimentrun.py
Comment (liuverta) [inline client/verta/verta/tracking/entities/_experimentrun.py]: Sorry this is an extra request, but could we do this so it is more similar to the other `def workspace()`s?

```python
    @property
    def workspace(self):
        self._refresh_cache()

        msg = ProjectService_pb2.GetProjectById(id=self._msg.project_id)
        url = "/api/v1/modeldb/project/getProjectById"
        response = self._conn.make_proto_request("GET", url, params=msg)
        response = self._conn.must_proto_response(response, msg.Response)
        proj_proto = response.project

        if proj_proto.workspace_service_id:
            return self._conn.get_workspace_name_from_id(proj_proto.workspace_service_id)
        else:
            return self._conn._OSS_DEFAULT_WORKSPACE
```

(will require a `from verta._protos.public.modeldb import ProjectService_pb2` at the top of the file)

File: client/verta/verta/client.py
Comment (liuverta) [reply client/verta/verta/client.py:1745]: > So as I see some of permission tests in Python Client can't be migrated to v2.

Yes, we will need to end up deleting tests that are no longer applicable to v2 😞

File: client/verta/verta/dataset/entities/_dataset.py
Comment (liuverta) [inline client/verta/verta/dataset/entities/_dataset.py]:
```suggestion
        if self._msg.workspace_service_id:
```

File: client/verta/verta/tracking/entities/_project.py
Comment (liuverta) [inline client/verta/verta/tracking/entities/_project.py]:
```suggestion
        if self._msg.workspace_service_id:
```

File: client/verta/verta/_internal_utils/_utils.py
Comment (liuverta) [reply client/verta/verta/_internal_utils/_utils.py:302]: Ah, I see entities have `workspace_service_id`. Okay, this sounds good to me 👍

**Review by liuverta — CHANGES_REQUESTED** [review]

File: client/verta/verta/_internal_utils/_utils.py
Comment (evgeniilezhnin) [reply client/verta/verta/_internal_utils/_utils.py]: yes, a PR for that is merged (https://vertaai.atlassian.net/browse/VUMM-808) so reverted this change
update: sorry, wrong ticket. The right one is https://vertaai.atlassian.net/browse/VUMM-809 . Not merged, but expected to be merged soon

File: client/verta/verta/_internal_utils/_utils.py
Comment (evgeniilezhnin) [reply client/verta/verta/_internal_utils/_utils.py]: reverted

File: client/verta/verta/tracking/entities/_experimentrun.py
Comment (evgeniilezhnin) [reply client/verta/verta/tracking/entities/_experimentrun.py]: applied your code snippet

File: client/verta/tests/test_permissions/test_workspace.py
Comment (liuverta) [inline client/verta/tests/test_permissions/test_workspace.py]: If possible, these three workspaces need to be different, not the same one.

File: client/verta/verta/_internal_utils/_utils.py
Comment (liuverta) [inline client/verta/verta/_internal_utils/_utils.py:270]: Just to double check: In permissions V2, `workspace.org_name` is still the accurate workspace name?

File: client/verta/tests/registry/model_version/test_from_run.py
Comment (liuverta) [inline client/verta/tests/registry/model_version/test_from_run.py]: Do you know if it is possible to keep these tests for V2?

**Review by liuverta — CHANGES_REQUESTED** [review]
Just a few more questions!

File: client/verta/tests/test_permissions/test_workspace.py
Comment (evgeniilezhnin) [reply client/verta/tests/test_permissions/test_workspace.py]: yes, it is possible, but we need workspace creation for that, so I added it now to workspace creation PR

File: client/verta/verta/_internal_utils/_utils.py
Comment (evgeniilezhnin) [reply client/verta/verta/_internal_utils/_utils.py:270]: yes

File: client/verta/tests/registry/model_version/test_from_run.py
Comment (evgeniilezhnin) [reply client/verta/tests/registry/model_version/test_from_run.py]: yes, it is possible, but we need workspace creation for that, so I added it now to workspace creation PR

File: client/verta/tests/test_permissions/test_workspace.py
Comment (liuverta) [reply client/verta/tests/test_permissions/test_workspace.py]: 👍 Great, I see that it was added in https://github.com/VertaAI/modeldb/pull/3439/commits/33079e293fc9ae1d70ee63bbd8389012e184c6f9

File: client/verta/tests/registry/model_version/test_from_run.py
Comment (liuverta) [reply client/verta/tests/registry/model_version/test_from_run.py]: 👍 Great, I see that it was added in https://github.com/VertaAI/modeldb/pull/3439/commits/29377da62da94712b41db213ebbd1612c12ae478

**Review by liuverta — APPROVED** [review]

**Pull request merged**
